src/extensions/query-translator/clients/OpenAiClient.ts

```diff
@@ -0,0 +1,155 @@
+import { ChatCompletionRequestMessage, ChatCompletionRequestMessageRoleEnum, Configuration, OpenAIApi } from 'openai';
+import { reportTypesToDesc, reportExampleQueries } from './const';
+import { ModelClient } from './ModelClient';
+import { Status } from '../component/ClientSettings';
+
+const consoleLogAsync = async (message: string, other?: any) => {
+  await new Promise((resolve) => setTimeout(resolve, 0)).then(() => console.info(message, other));
+};
+
+export class OpenAiClient extends ModelClient {
+  modelType: string | undefined;
+
+  createSystemMessage: any;
+
+  modelClient!: OpenAIApi;
+
+  driver: any;
+
+  constructor(settings) {
+    super(settings);
+  }
+
+  async validateQuery(message, database) {
+    let query = message.content;
+    let isValid = false;
+    let errorMessage = '';
+    try {
+      let res = await this.queryDatabase(`EXPLAIN ${query}`, database);
```

Comment:
would there be any way to 'escape' the explain and do some kind of query injection here?
Just theorizing if there's a possiblity of edits being done to the DB here

Reply:
I really don't know, but how the can comment out the EXPLAIN ?